cli/src/utils/cli-flags.ts

```diff
@@ -3,21 +3,23 @@ import { error } from '@oclif/errors';
 import { isISO8601, isInt } from 'validator';
 import ms from 'ms';
 
-export type MinMaxIntFlagOptions = Parameters<(typeof flags.integer)>[0] & {
+export type MinMaxIntFlagOptions = Parameters<(typeof flags.integer)>[0] & Partial<{
   min: number,
   max: number,
-};
+}>;
 
 export default {
   ...flags,
-  minMaxInt: ({ min, max }: MinMaxIntFlagOptions) => flags.integer({
+  minMaxInt: (options: MinMaxIntFlagOptions) => flags.build({
+    ...options,
     parse(val) {
+      const { min, max } = options;
       if (!isInt(val, { min, max })) {
         error(`Expected an integer between ${min} and ${max} but received: ${val}`);
       }
       return parseInt(val, 10);
     },
-  }),
+  })(),
```

Comment:
I don't understand this.  It looks like you're calling the return value of `flags.build` as a function with no args, and using the return of _that_ function as the value of `minMaxInt`.  How does this even work (or compile, assuming reasonable types)?

Or, why do you need to call the return value from `flags.build` here, but not below in `durationOrISO8601`?

Reply:
Yuck.  Thanks for explaining!